### PR TITLE
fixed + or - after factorial being ignored

### PR DESCRIPTION
**Pull request by mtettmar**
This fixes the following problem:

5*3!+2 evaluate to 30, but should be 32.  The +2 is being ignored.

Currently we have to use` (5*3!)+2 but 5*3!+2 `should be accepted etc.

This pull requests fixes that.

### CODE REVIEW

File: BigEval.js [hunk 1]
```diff
@@ -570,6 +570,12 @@ BigEval.prototype.compile = function (expression) {
 			((i > 1 && tokens[i - 2].type === TokenType.OP) || i === 1)
 			) {
 
+			console.log(tokens[i - 2]);
+
+			if (i > 1 && tokens[i - 2].value==='!') {
```
Comment (aviaryan) [inline BigEval.js:575]: If i = 2, wouldn't `!` being at token 0 be an invalid case?

I haven't looked at the code in years, so I apologize if I am wrong.

Also, can you please remove the console.log? Thanks.

File: BigEval.js [hunk 2]
```diff
@@ -570,6 +570,12 @@ BigEval.prototype.compile = function (expression) {
 			((i > 1 && tokens[i - 2].type === TokenType.OP) || i === 1)
 			) {
 
+			console.log(tokens[i - 2]);
+
+			if (i > 1 && tokens[i - 2].value==='!') {				
+				continue
+			}
+
```
Comment (aviaryan) [inline BigEval.js:578]: Adding a test for this fix would be nice too. We have automated tests if you pull the latest commits from `master`.

Can you pull the latest commits from `master` and add a test or two? Thanks.

File: BigEval.js (hunk 1 above)
Comment (mtettmar) [reply BigEval.js:640]: Oops. Sorry. That should be sorted now.

Comment (mtettmar) [reply BigEval.js:640]: Re I=2 trouble is ! can also be used as Not operator .. ?

File: BigEval.js (hunk 2 above)
Comment (mtettmar) [reply BigEval.js:643]: Have added a test.

**Comment by mtettmar** [comment]
Just fixed another issue. This was causing an unhandled execution, thus resulting in 'ERROR':

			if (!left && op === '-') {
				left = [{ type: TokenType.NUMBER, value: 0, source: expression }];
			}

expression does not exist. Removed source: expression to fix.

E.g.:

0.125^(-2/3) - gave 'ERROR' instead of 3.99999...

File: BigEval.js (hunk 1 above)
Comment (aviaryan) [reply BigEval.js]: You are right. I forgot we had NOT operator.

**Comment by aviaryan** [comment]
@mtettmar Merging #25 added conflicts to this MR. Can you please look into them?

**Comment by mtettmar** [comment]
Yep no problem - have done that.

**Comment by mtettmar** [comment]
Closing - now fixed in #26

**Pull request closed without merging**